agent/worker.go

```diff
@@ -31,6 +32,9 @@ type Worker interface {
 	//
 	// The listener will be removed if the context is cancelled.
 	Listen(ctx context.Context, reporter StatusReporter)
+
+	// GetSecret returns a secret from the local cache
+	GetSecret(id string) (*api.Secret, error)
```

Comment:
@diogomonica That is below. From where this is, I can't really tell what the actual use case is. What is accessing these? Why isn't there a secrets interface?